### DISCUSSION TITLE
Add Events to Calendar in Panel

### DISCUSSION THREAD

**Discussion by matdave** [pop-os/cosmic-epoch#1339] [Ideas]
I like having the dropdown calendar in the panel, but it seems relatively useless at the moment. 

![image](https://github.com/user-attachments/assets/6d50df82-1a3a-4777-817c-ff5ac5e0410c)

There's not really a need for a constant large button text that takes you to date and time settings. I don't think people are going to adjust those regularly. 

It would be really nice to be able to attach this to my google calendar to see events on each day. I find this feature extremely helpful on Gnome.

**Comment by GenericConfluent** [comment]
The settings button is common to a bunch of the applets. I'm kind-of mixed on my opinion of it. On the one hand, I use it for some widgets like network, input, and sound sometimes and it is a nice time save. On the other I never use the button on this particular applet so it ends up a waste of space. 

To me it's completely useless but feels right in terms of consistency. 

I've also kind-of wanted some calendar integration, just to be able to read ical and display the stuff in a list for the day. And I would expect that stuff to go into the time applet. Currently the time applet is the same width as all the others (since it uses `popup_container` the width is set to 360px). Working within that you could do something like this: 

<img width="367" height="759" alt="image" src="https://github.com/user-attachments/assets/77732a03-5a88-41fc-8b0d-5b2127d83318" />

That feels pretty unbalanced having a giant pillar going up and down the middle of the screen when the panel is at top or bottom and the widget it in the center position, but it looks decent in the screen corners or when the panel is on the right or left. 

Probably for a proper implementation what you would want to do is make the event list basically 
just normal text (changing the text so its colored like the calendar days are) and then highlighting the row with the current event if there is one. And then to deal with the fact that some people might not like the giant stack, add some options to "Date, Time, and Calendar Settings" to allow the user to select a vertical split or horizontal split, and select the event density (compact or relaxed).

**Comment by zothem** [comment]
i have already developed a basic version of this and have it working.  if there is interest behind it ill make a repo available to the community.

**Reply by GenericConfluent** [reply]
I've got some interest in this, but even more so I'm curious about the route you took to implement it.
